AutoHeuristic: Collect data for mixed_mm

AlnisM opened the pull request:
This PR introduces a script that can be used to collect data for mixed_mm to learn a heuristic with AutoHeuristic. This PR also includes the following things:

Move pad_mm related AutoHeuristic files into subdirectory
Introduce an interface benchmark_runner.py that can be subclassed to introduce new scripts to run benchmarks in order to collect data with AutoHeuristic (see gen_data_pad_mm.py and gen_data_mixed_mm.py).
The idea behind the interface is that, in the end, it hopefully makes it easier to collect data for new optimizations, and thus makes it easier to learn a heuristic.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132088
* #131714
* #131710
* #131705
* #131617
* #131616
* #131790
* #131615
* #131613
* __->__ #131611
* #131610

AlnisM committed:
AutoHeuristic: mixed_mm heuristic A100

[ghstack-poisoned]

AlnisM committed:
Update on "AutoHeuristic: Collect data for mixed_mm"

This PR introduces a script that can be used to collect data for mixed_mm to learn a heuristic with AutoHeuristic. This PR also includes the following things:

Move pad_mm related AutoHeuristic files into subdirectory
Introduce an interface benchmark_runner.py that can be subclassed to introduce new scripts to run benchmarks in order to collect data with AutoHeuristic (see gen_data_pad_mm.py and gen_data_mixed_mm.py).
The idea behind the interface is that, in the end, it hopefully makes it easier to collect data for new optimizations, and thus makes it easier to learn a heuristic.




[ghstack-poisoned]

eellison reviewed (approved):
For the trained heuristic you used on the dashboard, did you also include any shapes from timm/hf/torchbench, or was it all generated this way ?

eellison commented on torchgen/autoheuristic/benchmark_utils.py:
```diff
@@ -0,0 +1,47 @@
+
+
+def fits_in_memory(
+    dtype: Any, m: int, k: int, n: int, threshold_memory: int = 21474836480
```
we should be able to just query this. cuda_device_properties.total_memory

AlnisM committed:
Update on "AutoHeuristic: Collect data for mixed_mm"

This PR introduces a script that can be used to collect data for mixed_mm to learn a heuristic with AutoHeuristic. This PR also includes the following things:

Move pad_mm related AutoHeuristic files into subdirectory
Introduce an interface benchmark_runner.py that can be subclassed to introduce new scripts to run benchmarks in order to collect data with AutoHeuristic (see gen_data_pad_mm.py and gen_data_mixed_mm.py).
The idea behind the interface is that, in the end, it hopefully makes it easier to collect data for new optimizations, and thus makes it easier to learn a heuristic.




[ghstack-poisoned]

AlnisM commented:
@pytorchbot merge

AlnisM committed:
Update on "AutoHeuristic: Collect data for mixed_mm"

This PR introduces a script that can be used to collect data for mixed_mm to learn a heuristic with AutoHeuristic. This PR also includes the following things:

Move pad_mm related AutoHeuristic files into subdirectory
Introduce an interface benchmark_runner.py that can be subclassed to introduce new scripts to run benchmarks in order to collect data with AutoHeuristic (see gen_data_pad_mm.py and gen_data_mixed_mm.py).
The idea behind the interface is that, in the end, it hopefully makes it easier to collect data for new optimizations, and thus makes it easier to learn a heuristic.




[ghstack-poisoned]

AlnisM commented:
@pytorchbot merge

AlnisM commented:
@pytorchbot merge

AlnisM commented:
@pytorchbot merge
